### PR TITLE
Clean up more of SERVER_PROCESS

### PR DESCRIPTION
**Pull request by adityamandaleeka**
This is more code cleanup and should have no functional difference. More to come, but I don't want the PRs to get too large to review.

@BrennanConroy

### CODE REVIEW

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp [hunk 1]
```diff
@@ -106,29 +98,37 @@ SERVER_PROCESS::GetRandomPort
     DWORD  dwExcludedPort = 0
 )
 {
-    HRESULT hr = S_OK;
-    BOOL    fPortInUse = FALSE;
-    DWORD   dwActualProcessId = 0;
+    if (pdwPickedPort == nullptr)
```
Comment (BrennanConroy) [inline src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp:101]: This one isn't obvious to me, why the change?

Comment (adityamandaleeka) [reply src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp:101]: I've reconsidered this and turned it into a DBG_ASSERT now

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp [hunk 2]
```diff
@@ -54,49 +51,44 @@ SERVER_PROCESS::Initialize(
...
     {
-        goto Finished;
+        return hr;
     }
 
     m_pEnvironmentVarTable = pEnvironmentVariables;
 
-Finished:
-    return hr;
+    return S_OK;
```
[the remark this answers is not in the document]
Comment (adityamandaleeka) [reply src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp:59]: That's true, but the only caller today just uses `RETURN_IF_FAILED` on the result so it doesn't matter.

**Review by BrennanConroy — APPROVED** [review]
